api/v1alpha1/watcher_types.go

```diff
@@ -20,6 +20,10 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	WatcherControllerName = "watcher"
```

Comment:
I think it was a bad idea of me to originally include lifecycle manager in the API types. Ideally we should not add more controllers to the API types I believe.

Reply:
OK, I did that because I've seen this pattern, although I also found it strange. I will move it somewhere outside the API package.